Mathlib/Combinatorics/SetFamily/Compression/Down.lean

```diff
@@ -136,6 +136,81 @@ theorem nonMemberSubfamily_nonMemberSubfamily :
   simp
 #align finset.non_member_subfamily_non_member_subfamily Finset.nonMemberSubfamily_nonMemberSubfamily
 
+lemma memberSubfamily_image_insert (h𝒜 : ∀ s ∈ 𝒜, a ∉ s) :
+    (𝒜.image $ insert a).memberSubfamily a = 𝒜 := by
+  ext s
+  simp only [mem_memberSubfamily, mem_image]
+  refine ⟨?_, fun hs ↦ ⟨⟨s, hs, rfl⟩, h𝒜 _ hs⟩⟩
+  rintro ⟨⟨t, ht, hts⟩, hs⟩
+  rwa [←insert_erase_invOn.2.injOn (h𝒜 _ ht) hs hts]
+
+@[simp] lemma nonMemberSubfamily_image_insert : (𝒜.image $ insert a).nonMemberSubfamily a = ∅ := by
+  simp [eq_empty_iff_forall_not_mem]
+
+@[simp] lemma memberSubfamily_image_erase : (𝒜.image (erase · a)).memberSubfamily a = ∅ := by
+  simp [eq_empty_iff_forall_not_mem,
+    (ne_of_mem_of_not_mem' (mem_insert_self _ _) (not_mem_erase _ _)).symm]
+
+lemma image_insert_memberSubfamily (𝒜 : Finset (Finset α)) (a : α) :
+    (𝒜.memberSubfamily a).image (insert a) = 𝒜.filter (a ∈ ·) := by
+  ext s
+  simp only [mem_memberSubfamily, mem_image, mem_filter]
+  refine ⟨?_, fun ⟨hs, ha⟩ ↦ ⟨erase s a, ⟨?_, not_mem_erase _ _⟩, insert_erase ha⟩⟩
+  · rintro ⟨s, ⟨hs, -⟩, rfl⟩
+    exact ⟨hs, mem_insert_self _ _⟩
+  · rwa [insert_erase ha]
+
+/-- Induction principle for finset families. To prove a statement for every finset family,
+it suffices to prove it for
+* the empty finset family.
+* the finset family which only contains the empty finset.
+* `𝒜 ∪ {s ∪ {a} | s ∈ ℬ}` assuming the property for `𝒜` and `ℬ`, where `a` is an element of the
+  ground type and `𝒜` and `ℬ` are families of finsets not containing `a`.
```

Comment:
I find this docstring rather confusing; there are two different `𝒜`s in the theorem statement, but as far as I can tell neither is the one mentioned here.

Reply:
The first `𝒜` is just because it's an `_on` induction principle. The second `𝒜` is the one in the docstring.

Reply:
The goal is to restate the mathematics in a way that's closer to the textbook formulation, so I don't know how much I care about the mismatch.

Reply:
What about now?